alsa_settings/CAVS_NOCODEC_IPC4.sh

```diff
@@ -0,0 +1,15 @@
+set -e
+
+# SSP playback
+amixer -c sofnocodec cset name='gain.1.1 Playback Volume 1' 45
+amixer -c sofnocodec cset name='gain.15.1 Deepbuffer Volume' 45
+amixer -c sofnocodec cset name='gain.2.1 Main Playback Volume 2' 45
+amixer -c sofnocodec cset name='gain.3.1 Playback Volume 3' 45
+amixer -c sofnocodec cset name='gain.4.1 Main Playback Volume 4' 45
+amixer -c sofnocodec cset name='gain.5.1 Playback Volume 5' 45
+amixer -c sofnocodec cset name='gain.6.1 Main Playback Volume 6' 45
```

Comment:
Can you remind me why we should care about the volume settings in the firmware? This should be 0dB by default, always. the way this is implemented will break for every topology change or rename of controls.

Reply:
Yes, the default volume should be 0dB, but we cannot ensure that these volumes will not be modified before testing alsabat.

Reply:
@keqiaozhang you mean because other tests may change them? Why?

I've seen a large number of discussions about ALSA settings and UCM for a few years now and I still don't know how the "ideal state" should look like. Exceptions and quirks are sometimes necessary but I still can't tell what is a quirk versus what is a Good Thing. Is this already documented somewhere or would a new https://github.com/thesofproject/sof-test/discussions help?

Reply:
> Yes, the default volume should be 0dB, but we cannot ensure that these volumes will not be modified before testing alsabat.

then use a script to find all gain/PGA for the firmware using regexps and reset the value separately from codec stuff.

Reply:
> then use a script to find all gain/PGA for the firmware using regexps and reset the value separately from codec stuff

Can we merge this PR first? I will try your suggestion next. It's not only related to volume, but also related to switch.

Reply:
> then use a script to find all gain/PGA for the firmware using regexps and reset the value separately from codec stuff.

@keqiaozhang is this what you're trying to do in #1010?

Reply:
> @keqiaozhang is this what you're trying to do in #1010?

Yes, I'm working on it.